### PR TITLE
chore(RHTAPWATCH-765): appstudio-pipeline can manage SpaceRequests

### PR DESCRIPTION
**Pull request by Omeramsc**
In order to create ephemeral namespaces for
testing konflux components, the appstudio-pipeline
service account needs to be able to manage (create, delete, get) the SpaceRequest custom resource.

~Related PR:  https://github.com/redhat-appstudio/infra-deployments/pull/3203~

### CODE REVIEW

**Comment by MatousJobanek** [comment]
@Omeramsc could you please clarify why do you need to put it in member operator deployment manifests? Why this couldn't be moved into infra-deployments?
As I see it, creation of some special ClusterRoles should be done outside of member-operator (in your case in infra-deployments). I don't see any reason for putting it here.

**Comment by MatousJobanek** [comment]
If you need to create some extra permissions for your SA, then it should be created along with you SA, not as part of the member-operator.

**Comment by Omeramsc** [comment]
Closing this for now as indeed it doesn't seem relevant.

**Comment by Omeramsc** [comment]
Re-opening this as it seems to be the right place for this configuration after all. 
In the cluster we can see The member operator creates the service account. here we can see the ClusterRole defines with member operator labels.
```
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: appstudio-pipelines-runner
  uid: 3937c85d-83df-4f92-ba61-45d57f013588
  resourceVersion: '1356805130'
  creationTimestamp: '2023-05-08T18:39:35Z'
  labels:
    olm.owner: toolchain-member-operator.v0.0.476-commit-5b1f7c1
  managedFields:
    - manager: catalog
      operation: Update
      apiVersion: rbac.authorization.k8s.io/v1
      time: '2024-02-07T02:26:27Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:olm.owner': {}
        'f:rules': {}
rules:
  - verbs:
      - get
    apiGroups:
      - ''
    resources:
      - secrets
  - verbs:
      - use
    apiGroups:
      - security.openshift.io
    resources:
      - securitycontextconstraints
    resourceNames:
      - appstudio-pipelines-scc
  - verbs:
      - get
      - list
    apiGroups:
      - appstudio.redhat.com
    resources:
      - enterprisecontractpolicies
```
      
same in the SA:
```
kind: ServiceAccount
apiVersion: v1
metadata:
  name: appstudio-pipeline
  namespace: rh-devprod-tenant
  uid: ffd8b7a7-a3c9-492a-bf67-84036e6b2c77
  resourceVersion: '1326439732'
  creationTimestamp: '2023-08-04T10:10:57Z'
  labels:
    toolchain.dev.openshift.com/provider: codeready-toolchain
    toolchain.dev.openshift.com/space: rh-devprod
  managedFields:
    - manager: member-operator
      operation: Update
      apiVersion: v1
      time: '2023-08-04T10:10:57Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:toolchain.dev.openshift.com/provider': {}
            'f:toolchain.dev.openshift.com/space': {}
    - manager: openshift-controller-manager
      operation: Update
      apiVersion: v1
      time: '2023-08-04T10:10:57Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:secrets':
          .: {}
          'k:{"name":"appstudio-pipeline-dockercfg-thjnt"}': {}
    - manager: manager
      operation: Update
      apiVersion: v1
      time: '2024-01-31T19:28:21Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:imagePullSecrets': {}
        'f:secrets':
          'k:{"name":"ingress"}': {}
secrets:
  - name: appstudio-pipeline-dockercfg-thjnt
  - name: ingress
imagePullSecrets:
  - name: appstudio-pipeline-dockercfg-thjnt
  - name: ingress
```
  
The member operator manages the SA created for all namespaces, correct?  
The permissions I need to add are relevant to each and every SA created by it. just as the member operator creates the original runner role.

**Comment by mfrancisc** [comment]
If we are talking about the `appstudio-pipeline` SA it is being created by the host operator in each `-tenant` namespace https://github.com/codeready-toolchain/host-operator/blob/62b23df9ceffc6450a9eea86f4148927205df482/deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml#L245-L264. 

While the cluster role `appstudio-pipelines-runner` seems managed by pipeline-service component. So I guess we could add the `SpaceRequest` cluster role there as well ?

**Comment by Omeramsc** [comment]
Sorry for the misleading title - the SA is `appstudio-pipeline`, not what was mentioned previously. my bad. 

After a long Slack discussion[1] it seems like currently this is the right place for this configuration.
PTAL @MatousJobanek @mfrancisc @alexeykazakov 

[1]: https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1707731805481749

File: config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml [hunk 1]
```diff
@@ -25,3 +25,12 @@ rules:
...
   - get
   - list
+- apiGroups:
+  - toolchain.dev.openshift.com
+  resources:
+  - spacerequests
+  verbs:
+  - get
+  - create
+  - delete
```
Comment (mfrancisc) [inline config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml:35]: do you need also the other verbs ? 

```suggestion
  - get
  - create
  - delete
  - patch
  - list
  - update
  - watch
```

**Review by mfrancisc — COMMENTED** [review]
Looks Good 👍 

I have one minor comment/question.

**Review by alexeykazakov — APPROVED** [review]

File: config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml (hunk 1 above)
Comment (mfrancisc) [reply config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml:35]: So you don't need to watch or update the SpaceRequest once created ?

**Review by mfrancisc — APPROVED** [review]
LGTM

**Pull request merged**